src/OrbitGl/App.cpp

```diff
@@ -1658,12 +1658,14 @@ void OrbitApp::UpdateAfterCaptureCleared() {
   PostProcessedSamplingData empty_post_processed_sampling_data;
   absl::flat_hash_map<CallstackID, std::shared_ptr<CallStack>> empty_unique_callstacks;
 
-  SetSamplingReport(empty_post_processed_sampling_data, empty_unique_callstacks);
+  if (sampling_report_ != nullptr) {
+    SetSamplingReport(empty_post_processed_sampling_data, empty_unique_callstacks);
+  }
```

Comment:
Is this part of the fix? I'm a bit confused by it, mostly because I'm not up to date with this code. When is `sampling_report_ != nullptr` if we just cleared the capture? Or when is it `sampling_report_ == nullptr` when we just cleared the capture? How can we end up in two different states after clearing a capture?

Reply:
That is actually enough to do the fix. Before this, we've set sampling report (no matter what it was before) to "empty" sampling report on a start of the capture. In capture_data_ the post processed data is still completely missing, and will only appear after the capture. In the disassemble method we had then a check for sampling_report_ being nullptr and if not access capture data's post processed data. So by this line, sampling_report_ was not null (but empty) and capture data was still not having the data -> crash.

Reply:
Alright, thanks for the explanation. Yes, a larger refactor would be needed...

Reply:
Agree, but tbh. I do have a couple of crashes on my stack right now, so lets fix this first and do better later.

Reply:
Of course, of course, I wasn't suggesting we should tackle it right now!